android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/CheckBoxFactory.java

```diff
@@ -232,14 +235,22 @@ public void run() {
                                 checkBox.setChecked(Boolean.parseBoolean(item.optString(JsonFormConstants.VALUE)));
                             }
                         });
-                    }
-                    //Preselect values if they exist
-                    try {
-                        if (finalCheckBoxValues != null && getCurrentCheckboxValues(finalCheckBoxValues).contains(item.getString(JsonFormConstants.KEY))) {
-                            checkBox.setChecked(true);
+                    } else {
+                        //Preselect values if they exist
+                        try {
+                            if (finalCheckBoxValues != null) {
+                                for (String checkBoxVal : Objects.requireNonNull(getCurrentCheckboxValues(finalCheckBoxValues))) {
+                                    if (checkBoxVal != null && checkBoxVal.startsWith("{")) {
+                                        JSONObject jsonObject = new JSONObject(checkBoxVal);
+                                        if (jsonObject.optString(JsonFormConstants.VALUE).equalsIgnoreCase(item.getString(JsonFormConstants.KEY)) || getCurrentCheckboxValues(finalCheckBoxValues).contains(item.getString(JsonFormConstants.KEY))) {
+                                            checkBox.setChecked(true);
+                                        }
+                                    }
```

Comment:
I think that there should be an else clause here to handle support for apps without widget-value-translation enabled